src/game/msg_builder.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <core/provider.pb.h>
+#include <core/error.pb.h>
+#include <uno/uno.pb.h>
+
+#include "card.h"
+#include "../common/types.h"
+
+namespace SailGame { namespace Game {
+
+using ::Uno::GameStart;
+using ::Uno::NotifyMsg;
+using ::Uno::Draw;
+using ::Uno::Skip;
+using ::Uno::Play;
+using Core::ErrorNumber;
+using Core::NotifyMsgArgs;
+using Core::ProviderMsg;
+using Common::ProviderMsgPtr;
+
+class MsgBuilder {
+public:
+    static ProviderMsgPtr CreateRegisterArgs(int seqId, const std::string &id,
+        const std::string gameName, int maxUsers, int minUsers);
+
+    static ProviderMsgPtr CreateNotifyMsgArgs(int seqId, ErrorNumber err, 
+        int roomId, int userId, NotifyMsg custom);
```

Comment:
&？

Reply:
actually pass-by-reference is not necessary here. currently (and by design), the `custom` parameter should come from other methods of `MsgBuilder`. so there is no copy or move due to RVO even if it's passed by value.